src/dos/dos_classes.cpp

```diff
@@ -308,6 +309,119 @@ bool DOS_PSP::SetNumFiles(uint16_t file_num)
 	return true;
 }
 
+static constexpr auto bytes_to_read = 1024;
+
+std::optional<std::string> DOS_PSP::GetEnvironmentValue(const std::string_view variable) const
+{
+	/* Walk through the internal environment and see for a match */
+	PhysPt env_read = PhysicalMake(GetEnvironment(), 0);
+
+	char env_string[bytes_to_read + 1];
+	if (variable.empty()) {
+		return {};
+	}
+
+	for (;;) {
+		MEM_StrCopy(env_read, env_string, bytes_to_read);
+		if (!env_string[0]) {
+			return {};
+		}
+		env_read += (PhysPt)(safe_strlen(env_string) + 1);
+		char* equal = strchr(env_string, '=');
```

Comment:
Avoid C-style string operations and pointer arithmetic in new code.

Reply:
Was this set to `resolved` by mistake?

Reply:
I think this code was just moved around, the commit message even says so. It's fine to leave it as this for now, then refactoring can happen in a later PR as this is already quite overloaded 😅 Let's not add more to it now.